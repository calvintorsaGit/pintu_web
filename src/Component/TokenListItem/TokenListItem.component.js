import React from "react";
import {ReactSVG} from 'react-svg'
import './style.css'
import {Table} from "semantic-ui-react";
import {IoCaretUpOutline, IoCaretDownOutline} from 'react-icons/io5';


const formatToRupiah = (number) => {
    return new Intl.NumberFormat('id-ID', {
        style: 'currency',
        currency: 'IDR'
    }).format(number).replace('Rp', 'Rp ')
}

const _renderSymbol = (props) => {
    return <ReactSVG className='wrapper' style={{color: props.color}}
                     src={props.logo}
    />
}

const _renderPriceChange = (value) => {
    return <div class='flex flex-row items-center'>
        {parseFloat(value) > 0 ? <IoCaretUpOutline color='green'/> : <IoCaretDownOutline color='red'/>}
        <span
            className={parseFloat(value) > 0 ? 'font-bold text-xl text-green-600' : 'font-bold text-xl text-red-600'}>
            {value}
        </span>
    </div>
}

const TokenListItem = (props) => {
    return <Table.Row>
        <Table.Cell scope="row">{_renderSymbol(props)}</Table.Cell>
        <Table.Cell><span class='font-bold text-xl'>{props.name}</span></Table.Cell>
        <Table.Cell><span class='text-gray-500 text-xl'>{props.currencySymbol}</span></Table.Cell>
        <Table.Cell><span class='font-bold text-xl'>{formatToRupiah(props.latestPrice)}</span></Table.Cell>
        <Table.Cell>{_renderPriceChange(props.day)}</Table.Cell>
        <Table.Cell>{_renderPriceChange(props.week)}</Table.Cell>
        <Table.Cell>{_renderPriceChange(props.month)}</Table.Cell>
        <Table.Cell>{_renderPriceChange(props.year)}</Table.Cell>
    </Table.Row>
}

export default TokenListItem
