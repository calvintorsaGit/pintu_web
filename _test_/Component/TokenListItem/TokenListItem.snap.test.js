import renderer from 'react-test-renderer';
import React from "react";

import TokenListItem from "../../../src/Component/TokenListItem/TokenListItem.component";

jest.mock('react-svg');

it('renders correctly', () => {
    const tree = renderer
        .create(<TokenListItem/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
