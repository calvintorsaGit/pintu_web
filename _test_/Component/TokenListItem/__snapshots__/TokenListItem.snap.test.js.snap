// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<tr
  className=""
>
  <td
    className=""
    scope="row"
  />
  <td
    className=""
  >
    <span
      class="font-bold text-xl"
    />
  </td>
  <td
    className=""
  >
    <span
      class="text-gray-500 text-xl"
    />
  </td>
  <td
    className=""
  >
    <span
      class="font-bold text-xl"
    >
      Rp NaN
    </span>
  </td>
  <td
    className=""
  >
    <div
      class="flex flex-row items-center"
    >
      <svg
        color="red"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": "red",
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M98 190.06l139.78 163.12a24 24 0 0036.44 0L414 190.06c13.34-15.57 2.28-39.62-18.22-39.62h-279.6c-20.5 0-31.56 24.05-18.18 39.62z"
        />
      </svg>
      <span
        className="font-bold text-xl text-red-600"
      />
    </div>
  </td>
  <td
    className=""
  >
    <div
      class="flex flex-row items-center"
    >
      <svg
        color="red"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": "red",
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M98 190.06l139.78 163.12a24 24 0 0036.44 0L414 190.06c13.34-15.57 2.28-39.62-18.22-39.62h-279.6c-20.5 0-31.56 24.05-18.18 39.62z"
        />
      </svg>
      <span
        className="font-bold text-xl text-red-600"
      />
    </div>
  </td>
  <td
    className=""
  >
    <div
      class="flex flex-row items-center"
    >
      <svg
        color="red"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": "red",
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M98 190.06l139.78 163.12a24 24 0 0036.44 0L414 190.06c13.34-15.57 2.28-39.62-18.22-39.62h-279.6c-20.5 0-31.56 24.05-18.18 39.62z"
        />
      </svg>
      <span
        className="font-bold text-xl text-red-600"
      />
    </div>
  </td>
  <td
    className=""
  >
    <div
      class="flex flex-row items-center"
    >
      <svg
        color="red"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": "red",
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M98 190.06l139.78 163.12a24 24 0 0036.44 0L414 190.06c13.34-15.57 2.28-39.62-18.22-39.62h-279.6c-20.5 0-31.56 24.05-18.18 39.62z"
        />
      </svg>
      <span
        className="font-bold text-xl text-red-600"
      />
    </div>
  </td>
</tr>
`;
